### PR TITLE
Add support to Marketplace

### PR DESCRIPTION
**Pull request by Bonafede**
Add support to marketplace.
PR requested changes already made.

### CODE REVIEW

**Review by AleGladchteinParser — APPROVED** [review]

**Review by Armando-Ibarra — APPROVED** [review]
LGTM

Thanks for your patient on this PR.

File: src/CheckoutSdk/Marketplace/MarketplaceClient.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                SdkAuthorization(), 
+                entityRequest,
+                cancellationToken);
+        }
+
+        public async Task<object> CreatePaymentInstrument(string entityId, MarketplacePaymentInstrument marketplacePaymentInstrument, CancellationToken cancellationToken = default)
+        {
+            CheckoutUtils.ValidateParams("marketplacePaymentInstrument", marketplacePaymentInstrument, "entityId",
+                entityId);
+
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/MarketplaceClient.cs:55]: Please remove this blank line

File: src/CheckoutSdk/Marketplace/MarketplaceFileType.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    public class MarketplaceFileType
+    {
+        public string Value { get; private set; }
+
+        private MarketplaceFileType(string type)
+        {
+            Value = type;
+        }
+
+        public static MarketplaceFileType Passport
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/MarketplaceFileType.cs:12]: What do you think about refactoring this and the remaining methods to `expression-bodied` properties? Like this:

`public static MarketplaceFileType Passport => new MarketplaceFileType("passport")`

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        public string BranchCode { get; set; }
+
+        public string Iban { get; set; }
+
+        public string Bban { get; set; }
+
+        public string SwiftBic { get; set; }
+
+        public Currency Currency { get; set; }
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:27]: Please add `?` to enums.

File: src/CheckoutSdk/Marketplace/MarketplaceClient.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            return await ApiClient.Post<object>(
+                BuildPath(MarketplacePath, EntitiesPath, entityId, InstrumentPath),
+                SdkAuthorization(),
+                marketplacePaymentInstrument,
+                cancellationToken);
+        }
+
+        public async Task<IdResponse> SubmitFile(MarketplaceFileRequest marketplaceFileRequest, CancellationToken cancellationToken = default)
+        {
```
Comment (rsaestrela) [inline src/CheckoutSdk/Marketplace/MarketplaceClient.cs:64]: Missing `CheckoutUtils.ValidateParams`

**Review by rsaestrela — CHANGES_REQUESTED** [review]

File: src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs (hunk 3 above)
Comment (Bonafede) [reply src/CheckoutSdk/Marketplace/MarketplacePaymentInstrument.cs:27]: Already review by Armando: this enum is in a request object and is mandatory on the documentation.

File: src/CheckoutSdk/Marketplace/MarketplaceFileType.cs (hunk 2 above)
Comment (Bonafede) [reply src/CheckoutSdk/Marketplace/MarketplaceFileType.cs:12]: Nice!

File: src/CheckoutSdk/Marketplace/DateOfBirth.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Checkout.Marketplace
+{
+    public class DateOfBirth
+    {
+        public int Day { get; set; }
```
[the remark this answers is not in the document]
Comment (Bonafede) [reply src/CheckoutSdk/Marketplace/DateOfBirth.cs:5]: Already review by Armando: this fields are in a request object and are mandatory on the documentation.

File: src/CheckoutSdk/CheckoutConfiguration.cs [hunk 6]
```diff
@@ -1,21 +1,46 @@
...
+        public CheckoutConfiguration(
+            SdkCredentials sdkCredentials,
+            Environment environment,
+            IHttpClientFactory httpClientFactory,
+            Environment? filesEnvironment = null)
+        {
+            Init(sdkCredentials, environment, httpClientFactory, filesEnvironment);
+        }
+
+        private void Init(SdkCredentials sdkCredentials,
```
[the remark this answers is not in the document]
Comment (Bonafede) [reply src/CheckoutSdk/CheckoutConfiguration.cs:30]: Already review with Armando, this need a refactor of all integration test. We are doing it on another task.

File: src/CheckoutSdk/CheckoutConfiguration.cs [hunk 7]
```diff
@@ -1,21 +1,46 @@
...
-        public Environment Environment { get; }
+        public Environment Environment { get; private set; }
 
-        public IHttpClientFactory HttpClientFactory { get; }
+        public IHttpClientFactory HttpClientFactory { get; private set; }
 
-        public CheckoutConfiguration(SdkCredentials sdkCredentials, Environment environment,
+        public CheckoutFilesConfiguration FilesApiConfiguration { get; private set; }
+
+        public CheckoutConfiguration(
```
[the remark this answers is not in the document]
Comment (Bonafede) [reply src/CheckoutSdk/CheckoutConfiguration.cs:13]: Already review with Armando, this need a refactor of all integration test. We are doing it on another task.

File: src/CheckoutSdk/Common/Address.cs [hunk 8]
```diff
@@ -12,6 +12,6 @@ public class Address
 
         public string Zip { get; set; }
 
-        public CountryCode? Country { get; set; }
+        public CountryCode Country { get; set; }
```
[the remark this answers is not in the document]
Comment (Bonafede) [reply src/CheckoutSdk/Common/Address.cs:15]: No, the request has mandatory country when you send the address in the documentation, so CountryCode is not nulleable. The optional is Address in each class that references

**Pull request merged**